src/data-processor.js

```diff
@@ -0,0 +1,82 @@
+module.exports = {processContributors, processContent, processBranches, processCommits,
+    processCommitComments, processBlob};
+
+function processContributors(raw) {
+    /*
+    Iterate through array. Save all contributor names to a list
+
+    Relevant data:
+        raw.data[<index>].login
+    */
+    let contributorsData = [];
+
+    for (var i = 0; i < raw.data.length; i++) {
+        contributorsData.push(raw.data[i].login);
+    }
+
+    console.log("Contributor data:");
+    console.log(contributorsData);
+}
+
+function processContent(raw) {
+    /*
+    Iterate through the array, and save in object paths (file names) and shas
+    eg. {<file name>: {sha: <sha>} }
+
+    Relevant data:
+        raw.data.tree[<index>].path
+        raw.data.tree[<index>].sha
+    */
+    let fileData = {};
+
+    for (var i = 0; i < raw.data.tree.length; i++) {
+        currObj = raw.data.tree[i];
+        if (currObj.type === "blob" && currObj.path.endsWith(".java")) {
+            let fileName = currObj.path.split("/").pop();
+            let fileSha = currObj.sha;
+            let fileMeta = {}
+
+            fileMeta["sha"] = fileSha;
+            fileData[fileName] = fileMeta;
+        }
+    }
+
+    console.log("File data:");
+    console.log(fileData);
+}
+
+function processBranches(raw) {
+    /*
+    Currently unsure of purpose
+    */
+}
+
+function processCommits(raw) {
```

Comment:
If doids spawn based on file creation in commits, then getContent and processContent may be unnecessary